mlflow_databricks_artifacts/utils/rest_utils.py

```diff
@@ -0,0 +1,276 @@
+        true in production.
+        If this is set to true ``server_cert_path`` must not be set.
+    :param client_cert_path: Path to ssl client cert file (.pem).
+        Sets the cert param of the ``requests.request``
+        function (see https://requests.readthedocs.io/en/master/api/).
+    :param server_cert_path: Path to a CA bundle to use.
+        Sets the verify param of the ``requests.request``
+        function (see https://requests.readthedocs.io/en/master/api/).
+        If this is set ``ignore_tls_verification`` must be false.
+    """
+
+    def __init__(
+        self,
+        host,
+        username=None,
+        password=None,
+        token=None,
+        ignore_tls_verification=False,
+        client_cert_path=None,
+        server_cert_path=None,
+    ):
+        if not host:
+            raise MlflowException(
+                message="host is a required parameter for MlflowHostCreds",
+                error_code=INVALID_PARAMETER_VALUE,
+            )
+        if ignore_tls_verification and (server_cert_path is not None):
+            raise MlflowException(
+                message=(
+                    "When 'ignore_tls_verification' is true then 'server_cert_path' "
+                    "must not be set! This error may have occurred because the "
+                    "'MLFLOW_TRACKING_INSECURE_TLS' and 'MLFLOW_TRACKING_SERVER_CERT_PATH' "
+                    "environment variables are both set - only one of these environment "
+                    "variables may be set."
+                ),
+                error_code=INVALID_PARAMETER_VALUE,
+            )
+        self.host = host
+        self.username = username
+        self.password = password
+        self.token = token
+        self.ignore_tls_verification = ignore_tls_verification
+        self.client_cert_path = client_cert_path
+        self.server_cert_path = server_cert_path
+
+
+def http_put(*args, retry_attempts=5, **kwargs):
+    """
+    Performs an HTTP PUT request using Python's `requests` module with an automatic retry 
+    policy of `retry_attempts` using exponential backoff for the following response codes:
+
+        - 408 (Request Timeout)
+        - 429 (Too Many Requests)
+        - 500 (Internal Server Error)
+        - 502 (Bad Gateway)
+        - 503 (Service Unavailable)
+        - 504 (Gateway Timeout)
+
+    :args: Positional arguments to pass to `requests.Session.put()`
+    :kwargs: Keyword arguments to pass to `requests.Session.put()`
+    """
+    retry_strategy = Retry(
+        total=0,
+        status=retry_attempts,
+        status_forcelist=TRANSIENT_FAILURE_RESPONSE_CODES,
+        backoff_factor=1,
+    )
+    adapter = HTTPAdapter(max_retries=retry_strategy)
+    http = requests.Session()
+    http.mount("https://", adapter)
+    http.mount("http://", adapter)
+    return http.put(*args, **kwargs)
+
+
+def http_get(*args, retry_attempts=5, **kwargs):
+    """
```

Comment:
These methods can probably be DRY'ed out a bit, but the essence is that they're designed to provide automatic retry support for common retryable HTTP responses (429s, 503s, etc).